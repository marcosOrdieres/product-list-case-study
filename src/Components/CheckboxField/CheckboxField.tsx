import React from 'react'
import styled from 'styled-components'

const CheckboxContainer = styled.div`
  width:270px;
`
const Label = styled.label``

const StyledCheckbox = styled.input``

export interface CheckboxProps {
  checked?: boolean,
  onChange?: any,
  textLabel?: string,
};

const Checkbox = (props: CheckboxProps) => (
  <CheckboxContainer>
    <StyledCheckbox type='checkbox' checked={props.checked} onChange={props.onChange} />
    <Label>
      {props.textLabel}
    </Label>
  </CheckboxContainer>
)

export default Checkbox
