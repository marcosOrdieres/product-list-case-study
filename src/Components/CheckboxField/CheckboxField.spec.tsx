import React from 'react';
import { render } from '@testing-library/react';

import CheckboxField from './CheckboxField';

describe('CheckboxField', () => {
  it('should render CheckboxField and match styles', () => {
    const { asFragment } = render(
      <CheckboxField textLabel={''} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
