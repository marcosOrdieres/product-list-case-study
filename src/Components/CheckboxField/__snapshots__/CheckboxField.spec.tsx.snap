// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField should render CheckboxField and match styles 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB boEdTa"
  >
    <input
      class="sc-dlfnuX"
      type="checkbox"
    />
    <label
      class="sc-gsTEea"
    />
  </div>
</DocumentFragment>
`;
